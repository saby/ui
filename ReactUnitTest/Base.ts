import { Control } from 'UICore/Base';
import { IControlOptions } from 'UICommon/Base';

export interface ITestOptions extends IControlOptions {
    testName?: string;
    afterMountCallback?: Function;
    afterUpdateCallback?: Function;
}

export class TestBaseControl<
    IOptions extends ITestOptions = ITestOptions
> extends Control<IOptions> {
    testName: string = '';
    afterMountCallback: Function = null;
    afterUpdateCallback: Function = null;
    _beforeMount(options: ITestOptions): void {
        this.afterMountCallback = options.afterMountCallback;
        this.afterUpdateCallback = options.afterUpdateCallback;
        this.testName = options.testName;
    }
    _afterMount(): void {
        this.afterMountCallback?.(this);
    }
    _afterUpdate(): void {
        this.afterUpdateCallback?.(this);
    }
}
