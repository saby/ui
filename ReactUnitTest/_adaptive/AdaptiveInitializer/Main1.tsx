import Layout1 from './Layout1';
import { AdaptiveInitializerConfig } from 'UICore/Adaptive';

export default function Main1() {
    return (
        <AdaptiveInitializerConfig isPhoneForced={true}>
            <Layout1 />
        </AdaptiveInitializerConfig>
    );
}
