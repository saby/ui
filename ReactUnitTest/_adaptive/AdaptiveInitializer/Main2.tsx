import Layout1 from './Layout1';
import { AdaptiveInitializerConfig } from 'UICore/Adaptive';

export default function Main2() {
    return (
        <AdaptiveInitializerConfig isPhoneForced={false}>
            <Layout1 />
        </AdaptiveInitializerConfig>
    );
}
