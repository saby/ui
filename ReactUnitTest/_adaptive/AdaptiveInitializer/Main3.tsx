import Layout1 from './Layout1';

export default function Main1() {
    return <Layout1 />;
}
