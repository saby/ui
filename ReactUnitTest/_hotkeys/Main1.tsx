import { KeyHook } from 'UI/HotKeys';
import { constants } from 'Env/Env';
import { useCallback, useState } from 'react';
import * as React from 'react';

export default React.memo(function Main1() {
    const [value, setValue] = useState(0);
    const keyDownHandler = useCallback(() => {
        setValue(value + 1);
    }, []);
    return (
        <div onKeyDown={keyDownHandler}>
            <KeyHook defaultActions={[{ keyCode: constants.key.enter }]}>
                <div>обработчик keydown сработал: {value}</div>
            </KeyHook>
        </div>
    );
});
