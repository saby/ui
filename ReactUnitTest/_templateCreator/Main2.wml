<ReactUnitTest._templateCreator.Hoc value="{{value}}">
   <ReactUnitTest._templateCreator.Child value="{{value}}"/>
</ReactUnitTest._templateCreator.Hoc>
