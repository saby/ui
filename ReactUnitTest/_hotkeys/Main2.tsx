import { Component } from 'react';
import Main1 from './Main1';

interface IState {
    className: string;
}

export default class Main2 extends Component<any, IState> {
    constructor(props: any) {
        super(props);
        this.state = { className: '' };
    }

    setClassName(className: string) {
        this.setState({ className });
    }
    render() {
        return (
            <div className={this.state.className}>
                <Main1 />
            </div>
        );
    }
}
