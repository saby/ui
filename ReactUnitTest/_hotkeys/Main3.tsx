import { Component } from 'react';
import Main1 from './Main1';

interface IState {
    name: string;
}

export default class Main2 extends Component<any, IState> {
    constructor(props: any) {
        super(props);
        this.state = { name: '' };
    }

    setContainer(name: string) {
        this.setState({ name });
    }
    render() {
        return this.state.name === 'article' ? (
            <article>
                <Main1 />
            </article>
        ) : (
            <div>
                <Main1 />
            </div>
        );
    }
}
