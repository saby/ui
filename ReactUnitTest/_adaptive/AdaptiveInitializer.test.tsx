/**
 * @jest-environment jsdom
 */
import { Storage } from 'UICore/_adaptive/Aspects';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import Main1 from './AdaptiveInitializer/Main1';
import Main2 from './AdaptiveInitializer/Main2';
import Main3 from './AdaptiveInitializer/Main3';
import { Control } from 'UI/Base';

describe('AdaptiveInitializer', () => {
    let container: HTMLDivElement;

    beforeEach(() => {
        container = document.createElement('div');
        document.body.appendChild(container);
    });
    afterEach(() => {
        Storage.getInstance()._clear();
        unmountComponentAtNode(container);
        container.remove();
        container = null;
        document.body.removeAttribute('style');
        Storage.getInstance()._clear();
        jest.restoreAllMocks();
    });

    test('isPhoneForced = true', () => {
        act(() => {
            render(<Main1 />, container);
        });
        expect(document.body).toMatchSnapshot();
    });

    test('isPhoneForced = false', () => {
        act(() => {
            render(<Main2 />, container);
        });
        expect(document.body).toMatchSnapshot();
    });

    test('without isPhoneForced', () => {
        act(() => {
            render(<Main3 />, container);
        });
        expect(document.body).toMatchSnapshot();
    });
});

describe('AdaptiveInitializer with special domElementWithSizes', () => {
    let container: HTMLDivElement;
    let mainContainer: HTMLDivElement;

    beforeEach(() => {
        Storage.getInstance()._clear();
        mainContainer = document.createElement('div');
        mainContainer.setAttribute('style', 'width:1000px;height:1000px;');
        container = document.createElement('div');
        mainContainer.appendChild(container);
        document.body.appendChild(mainContainer);
    });
    afterEach(() => {
        unmountComponentAtNode(container);
        container.remove();
        mainContainer.remove();
        container = null;
        mainContainer = null;
        document.body.removeAttribute('style');
        Storage.getInstance()._clear();
        jest.restoreAllMocks();
    });

    test('isPhoneForced = true', () => {
        act(() => {
            Control.createControl(Main1, {}, container);
        });
        expect(document.body).toMatchSnapshot();
    });

    test('isPhoneForced = false', () => {
        act(() => {
            Control.createControl(Main2, {}, container);
        });
        expect(document.body).toMatchSnapshot();
    });

    test('without isPhoneForced', () => {
        act(() => {
            Control.createControl(Main3, {}, container);
        });
        expect(document.body).toMatchSnapshot();
    });
});
