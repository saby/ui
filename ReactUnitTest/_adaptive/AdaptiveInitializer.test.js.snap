// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdaptiveInitializer isPhoneForced = false 1`] = `
<body>
  <div>
    <div>
      adaptiveMode: 
      {"$serialized$":"inst","module":"UICore/Adaptive:_AdaptiveModeClass","id":2,"state":{"containerClientWidth":0,"containerClientHeight":0}}
      domElementWithSizes: 
      isPhoneForced: 
      false
    </div>
  </div>
</body>
`;

exports[`AdaptiveInitializer isPhoneForced = true 1`] = `
<body>
  <div>
    <div>
      adaptiveMode: 
      {"$serialized$":"inst","module":"UICore/Adaptive:_AdaptiveModeClass","id":1,"state":{"containerClientWidth":0,"containerClientHeight":0}}
      domElementWithSizes: 
      isPhoneForced: 
      true
    </div>
  </div>
</body>
`;

exports[`AdaptiveInitializer with special domElementWithSizes isPhoneForced = false 1`] = `
<body>
  <div
    style="width:1000px;height:1000px;"
  >
    <div
      style="height: 100%; width: 100%; display: contents;"
    >
      <div>
        adaptiveMode: 
        {"$serialized$":"inst","module":"UICore/Adaptive:_AdaptiveModeClass","id":5,"state":{"containerClientWidth":0,"containerClientHeight":0}}
        domElementWithSizes: 
        width:1000px;height:1000px;
        isPhoneForced: 
        false
      </div>
    </div>
  </div>
</body>
`;

exports[`AdaptiveInitializer with special domElementWithSizes isPhoneForced = true 1`] = `
<body>
  <div
    style="width:1000px;height:1000px;"
  >
    <div
      style="height: 100%; width: 100%; display: contents;"
    >
      <div>
        adaptiveMode: 
        {"$serialized$":"inst","module":"UICore/Adaptive:_AdaptiveModeClass","id":4,"state":{"containerClientWidth":0,"containerClientHeight":0}}
        domElementWithSizes: 
        width:1000px;height:1000px;
        isPhoneForced: 
        true
      </div>
    </div>
  </div>
</body>
`;

exports[`AdaptiveInitializer with special domElementWithSizes without isPhoneForced 1`] = `
<body>
  <div
    style="width:1000px;height:1000px;"
  >
    <div
      style="height: 100%; width: 100%; display: contents;"
    >
      <div>
        adaptiveMode: 
        {"$serialized$":"inst","module":"UICore/Adaptive:_AdaptiveModeClass","id":6,"state":{"containerClientWidth":0,"containerClientHeight":0}}
        domElementWithSizes: 
        width:1000px;height:1000px;
        isPhoneForced: 
        undefined
      </div>
    </div>
  </div>
</body>
`;

exports[`AdaptiveInitializer without isPhoneForced 1`] = `
<body>
  <div>
    <div>
      adaptiveMode: 
      {"$serialized$":"inst","module":"UICore/Adaptive:_AdaptiveModeClass","id":3,"state":{"containerClientWidth":0,"containerClientHeight":0}}
      domElementWithSizes: 
      isPhoneForced: 
      undefined
    </div>
  </div>
</body>
`;
